cdk/cmr-stac/lib/cmr-stac-stack.ts

```diff
@@ -0,0 +1,260 @@
+// Default memory size for lambda func
+const memorySize = 1024
+
+// NodeJS bundling options
+const bundling = {
+  // Only minify in production
+  minify: true,
+  externalModules: ['@aws-sdk/*']
+}
+
+/**
+ * The AWS CloudFormation template for this Serverless application
+ */
+export class CmrStacStack extends cdk.Stack {
+  // public readonly serverlessDeploymentBucketName;
+  /**
+   * Current Lambda function version
+   */
+  public readonly stacLambdaFunctionQualifiedArn: string;
+
+  public constructor(scope: cdk.App, id: string, props: CmrStacStackProps = {}) {
+    super(scope, id, props);
+    const stacLogGroup = new logs.LogGroup(this, `StacLogGroup`, {
+      logGroupName: `/aws/lambda/cmr-stac-api-eudoro-${STAGE_NAME}-stac${logGroupSuffix}`,
+      retention: logs.RetentionDays.ONE_MONTH
+    })
+    
+    const cmrStacDocsBucket = new s3.CfnBucket(this, 'cmrStacDocsBucket', {
+      bucketName: `cmr-stac-static-bucket-eudoro-${STAGE_NAME}`,
+    })
+    
+    const destinationBucket = s3.Bucket.fromCfnBucket(cmrStacDocsBucket)
+    new s3Deployment.BucketDeployment(this, `cmrStacDocumentation`, {
+      destinationBucket: destinationBucket,
+      sources: [
+        s3Deployment.Source.asset('../../docs/index')
+      ],
+      include: ['*'],
+      cacheControl: [
+        s3Deployment.CacheControl.maxAge(cdk.Duration.days(365))
+      ],
+      destinationKeyPrefix: "docs/",
+      retainOnDelete: true,
+      prune: false
+    })
+
+    const cmrStacRole = new iam.CfnRole(this, 'cmrStacRole', {
+      roleName: `stacRole-eudoro-${STAGE_NAME}`,
+      permissionsBoundary: [
+        'arn:aws:iam::',
+        this.account,
+        ':policy/NGAPShRoleBoundary',
+      ].join(''),
+      managedPolicyArns: [
+        'arn:aws:iam::aws:policy/service-role/AWSLambdaVPCAccessExecutionRole',
+      ],
+      assumeRolePolicyDocument: {
+        Version: '2012-10-17',
+        Statement: [
+          {
+            Effect: 'Allow',
+            Principal: {
+              Service: [
+                'lambda.amazonaws.com',
+              ],
+            },
+            Action: [
+              'sts:AssumeRole',
+            ],
+          },
+        ],
+      },
+    })
+
+    const vpc = Vpc.fromVpcAttributes(this, 'Vpc', {
+      availabilityZones: ['us-east-1a', 'us-east-1b','us-east-1c'],
+      privateSubnetIds: [
+        SUBNET_ID_A!,
+        SUBNET_ID_B!,
+        SUBNET_ID_C!
+      ],
+      vpcId: VPC_ID!,
+    })
+
+    const lambdaSecurityGroup = SecurityGroup.fromSecurityGroupId(this, 'CmrStacSecurityGroup', CMR_SERVICE_SECURITY_GROUP_ID);
+    
+    // Environment variables to pass to application
+    const environment = {
+      CMR_URL,
+      CMR_LB_URL,
+      GRAPHQL_URL,
+      STAC_VERSION,
+      PAGE_SIZE,
+      LOG_LEVEL
+    }
+    // TODO can I make this lambda using the library
+    const stacLambdaFunction = new lambdaNodeJs.NodejsFunction(this, `StacLambdaFunction`, {
```

Comment:
I didn't use the `application` node lambda because in the original config that aws made with the template the output is returned from the stack is used so it was more similar to `earthdataVarinfoLambdaFunction` in graphql